pkg/network/protocols/telemetry/metric.go

```diff
@@ -104,22 +97,25 @@ func (m *Metric) MarshalJSON() ([]byte, error) {
 		Opts []string
 	}{
 		Name: m.name,
-		Tags: m.tags,
-		Opts: m.opts,
+		Tags: m.tags.List(),
+		Opts: m.opts.List(),
 	})
 }
 
 func (m *Metric) isEqual(other *Metric) bool {
-	if m.name != other.name || len(m.tags) != len(other.tags) {
-		return false
-	}
+	return m.name == other.name && m.tags.Equal(other.tags)
+}
 
-	// Tags are always sorted
-	for i := range m.tags {
-		if m.tags[i] != other.tags[i] {
-			return false
-		}
+func parseMetricType(opts sets.String) metricType {
+	defer func() {
+		// remove type parameters from the options set
+		opts.Delete(OptGauge)
+		opts.Delete(OptCounter)
```

Comment:
unsure why is it here

Reply:
Yeah, I'm going to remove these type arguments and have a `Counter` and `Gauge` type instead